src/Bridges/ApplicationDI/ApplicationExtension.php

```diff
@@ -56,10 +64,16 @@ public function loadConfiguration()
 			$application->addSetup('Nette\Bridges\ApplicationTracy\RoutingPanel::initializePanel');
 		}
 
-		$touchToRebuild = $this->debugMode && $config['scanDirs'] ? reset($config['scanDirs']) : NULL;
+		$code = '!($me = $this) ?: ' . $container->formatPhp(
+			'function ($class) use ($me) { return Nette\Bridges\ApplicationDI\Helpers::createPresenter($me, $class, ?, ?); }',
```

Comment:
I like the closure, but I don't like that it's static class, when it doesn't have to be static class.

you could have created

```php
class PresenterFactoryCallback
{
    public function __construct(/* deps */);
    public function __invoke($class);
}
```

and attach it like

```php
$presenterFactory = $container->addDefinition($this->prefix('presenterFactory'))
    ->setFactory('Nette\Application\PresenterFactory', [
        new Nette\DI\Statement('PresenterFactoryCallback')
    ])
```

IMHO a lot nicer, what do you think?